File: action-sync/0946-smallest-range-ii/solution.java
import java.util.Arrays;

class Solution {
    public int smallestRangeII(int[] nums, int k) {
        Arrays.sort(nums);
        int n = nums.length;

        int result = nums[n - 1] - nums[0]; // 初始范围：全都不动
        for (int i = 0; i < n - 1; i++) {
            int high = Math.max(nums[n - 1] - k, nums[i] + k); // 右边的减 k，左边的加 k
            int low = Math.min(nums[0] + k, nums[i + 1] - k);   // 左边的加 k，右边的减 k
            result = Math.min(result, high - low);
        }

        return result;
    }
}

File: action-sync/0946-smallest-range-ii/solution.py
class Solution:
    def smallestRangeII(self, nums: List[int], k: int) -> int:
        nums.sort()
        res = nums[-1] - nums[0]  # 初始范围
        for i in range(len(nums) - 1):
            high = max(nums[i] + k, nums[-1] - k)
            low = min(nums[0] + k, nums[i + 1] - k)
            res = min(res, high - low)
        return res

        
